### DISCUSSION TITLE
Generate the secrets bundle from existing install

### DISCUSSION THREAD

**Discussion by Ulrar** [siderolabs/talos#7887] [Q&A]
Hi,

I followed the getting started guide and successfully started a new cluster, and only later noticed the note in the production installation notes advising to first generate a secrets bundle.

Is there any way to extract it from the running cluster, or do I need to throw it away and start over ?
I had to do quite a bit of manual edits to the controlplane.yaml file so I'd rather keep it since it's running, but from the docs it does look like having that secrets.yaml file would be handy in the future.

Thanks

**Comment by smira** [comment]
there's no automated way as far as I know at the moment, even though it should be easy to do so.

you can pull back secrets from `controlplane.yaml` and build a `secrets.yaml` from it. this is worth an issue to add support for it in `talosctl gen secrets`

**Reply by Ulrar** [reply]
Figured as much, opened #7895 for it, thanks !
